z3/src/z3_datatype.cpp

```diff
@@ -0,0 +1,40 @@
+#include "z3_datatype.h"
```

Comment:
This file is also pretty different from the cvc5 version, which is way shorter. Is that needed?

Reply:
Done. I kept only the compare function and the auxiliary functions for it.